chapter3.adoc

```diff
@@ -62,16 +40,50 @@ The field `relaxedpriv` in abstractcs (at 0x16) allows for relaxed permission ch
+
+[NOTE]
+In scenarios where a Debug Module lacks System Bus Access, memory access by the debugger can be achieved through the use of abstract commands. These commands provide a secure means of accessing memory.
 
 [NOTE]
 Trusted entities like RoT should configure IOPMP or equivalent protection before granting debug access to machine mode. Similarly, machine mode should apply the protection before enabling submachine mode debug. 
 
 === Security Fault Error Reporting
 
-A dedicated error code, security fault error (cmderr 6), is included in cmderr of abstractcs (at 0x16) to signal any breaches of security enforcement by debug operations. Additionally, the bus security fault error (sberror 6) is introduced in sberror of sbcs (at 0x38) to denote errors related to system bus access.
+A dedicated error code, security fault error (cmderr 6), is included in cmderr of abstractcs (at 0x16) to signal any breaches of security enforcement by abstract commands. Additionally, the bus security fault error (sberror 6) is introduced in sberror of sbcs (at 0x38) to denote errors related to system bus access. The error raised by other debug operations which could be applied to multiple harts such as reset, keepalive can be determined by the fields allsecfault/anysecfault in dmstatus.
+
+
+=== Update of Debug Module Status (dmstatus, at 0x11)
+
+[caption="Register {counter:rimage}: ", reftext="Register {rimage}"]
+[title="The `allsecured` and `anysecured` fields in dmstatus"]
+[id=dmstatus]
+[wavedrom, ,svg]
+....
+{reg: [
+  {bits:   20, name: 'defined in Debug Module'},
+  {bits:   1, name: 'allsecured'},
+  {bits:   1, name: 'anysecured'},
+  {bits:   3, name: 'defined in Debug Module'},
+  {bits:   1, name: 'allsecfault'},
+  {bits:   1, name: 'anysecfault'},
+  {bits:   5, name: '0'},
+], config:{lanes: 3, hspace:1024}}
+....
+
+[[regdmstatus]]
+.The updates in dmstatus 
+[options="header"]
+|================================================================================================================================================
+| Field       | Description                                                                                                      | Access | Reset
+| allsecured  | The field is 1 when all currently selected harts are secured and not running in debuggable privilege levels      | R      | -    
+| anysecured  | The field is 1 when any currently selected hart is secured and not running in debuggable privilege levels        | R      | -    
+| allsecfault | The field is 1 when all currently selected harts have raised security fault due to reset or keepalive operation. | R      | -    
```

Comment:
It is not defined how to clear anysecfault and allsecfault. There should be a way to clear them similar to other dmstatus register fields. For example dmstatus.havereset fields are cleared by writing to dmcontrol.ackhavereset. My proposals for clearing:

- Follow similar mechanism and acksecfault bit somewhere (note that there is no available bit in dmcontrol)
- define allsecured and anysecured bits in dmstatus as cleared when read (this clearing mechanim is not used in Debug Spec so far)
- define allsecured and anysecured bits in dmstatus as cleared when written (dmstatus fields are read-only in Debug Spec so far)

Reply:
The error reporting in dmstatus is only effective for operations: reset, resethaltreq and keepalive. What about we defining that reading anysecfault/anysecfault will always get the error status of the last operation? That is, the error will be only cleared after a successful operation.

Reply:
I like your idea if we assume updating the status bits (secfault) does not take arbitrary long time and it will be available for the next DMI read. If we cannot guarantee that there is no way to understand the status bits corresponds to the last operation. If we put this requirement in the specification, I am fine with your proposal. 
In general, there is no dependency between DMI operations. For example, when Debugger SW request hartreset (dmcontrol.hartreset=1), it takes arbitrary long time to reset the selected harts. Debugger SW needs to poll dmcontrol.havereset bits.

Reply:
@AoteJin If we can converge for this issue, I am ready to approve the PR. This issue is blocking for me.

Reply:
"The security fault errors are required to be detectable before the following read of the register responsible for reporting the error. "

Do you think adding above statement help mitigate the problem you mentioned?

Reply:
Yes, adding above statement will mitigate the problem. Debugger SW can rely on this requirement and It is feasible to meet this requirement in RTL implementation. Please be precise to avoid confusion with other security fault errors (abstract command, System Bus Security fault errors)

Reply:
Thanks @vicky-goode-img to share the thoughts. You made the points very much clear. I would say  option 1 is more intuitive to me. I will update the text to reflect the behavior of option 1.

Reply:
Thanks @AoteJin, the update looks good to me.

Reply:
I am fine with the updated description.